### ISSUE TITLE
Toolbar not clickable -- needs upgrade to "leaflet-draw-toolbar" package (next steps outlined)

### ISSUE THREAD

**Issue by rakeshtinku** [publiclab/Leaflet.DistortableImage#78]
In previous version of leaflet(0.7.7) toolbar is enabled when clicked on image but with the leaflet 1.2 version there are few issues. The click events are not registered in _initEvents(); , and when we click on image overlay this._showToolbar() is not invoking.

```
leaflet.js:5 Deprecated include of L.Mixin.Events: this property will be removed in future releases, please inherit from L.Evented instead. Error
    at y (file:///home/gaian/nodejs/Leaflet.DistortableImage/node_modules/leaflet/dist/leaflet.js:5:2048)
    at Function.v.extend (file:///home/gaian/nodejs/Leaflet.DistortableImage/node_modules/leaflet/dist/leaflet.js:5:14706)
    at file:///home/gaian/nodejs/Leaflet.DistortableImage/node_modules/leaflet-toolbar/dist/leaflet.toolbar.js:1:60
    at file:///home/gaian/nodejs/Leaflet.DistortableImage/node_modules/leaflet-toolbar/dist/leaflet.toolbar.js:1:5309 
```

**Comment by rakeshtinku** [comment]
i want to work on this can you guide me

**Comment by jywarren** [comment]
Yes, I believe this is due to upstream issues with https://github.com/Leaflet/Leaflet.toolbar/ not being compatible with Leaflet `v1.x`

I'm not as familiar with that but I believe it's a simpler library than this and I've posted at least one "deprecation" error which may be related. I've documented the error i see, and I hope we can narrow down what's going wrong soon: 

https://github.com/Leaflet/Leaflet.toolbar/issues/41

Thank you very much! We'd love to see this fixed.

**Comment by jywarren** [comment]
This has been fixed in the newer [leaflet-draw-toolbar](https://github.com/justinmanley/leaflet-draw-toolbar)

Usage example is here: https://github.com/justinmanley/leaflet-draw-toolbar/blob/master/examples/popup.html#L69-L71

```js
				new LeafletToolbar.EditToolbar.Popup(event.latlng, {
					actions: editActions
				}).addTo(map, layer);
```

Very similar to usage in this library: https://github.com/publiclab/Leaflet.DistortableImage/blob/master/src/edit/DistortableImage.EditToolbar.js#L95-L104

```js

L.DistortableImage.EditToolbar = L.Toolbar.Popup.extend({
	options: {
		actions: [
			ToggleTransparency,
			RemoveOverlay,
			ToggleOutline,
			ToggleEditable,
			ToggleRotateDistort
		]
	},

	/* Remove the toolbar after each action. */
	_getActionConstructor: function(Action) {
		var A = Action.extend({
			removeHooks: function() {
				var map = this._map;

				map.removeLayer(this.toolbar);
			}
		});

		return L.Toolbar.prototype._getActionConstructor.call(this, A);
	}
});
```

Although i'm not sure about that extra code to remove the toolbar after each action. May or may not be necessary still. 

I think the key difference will be how the actions are implemented. Currently we [do this](https://github.com/publiclab/Leaflet.DistortableImage/blob/master/src/edit/DistortableImage.EditToolbar.js#L3-L10):

```js
EditOverlayAction = L.ToolbarAction.extend({
		initialize: function(map, overlay, options) {
			this._overlay = overlay;
			this._map = map;

			L.ToolbarAction.prototype.initialize.call(this, options);
		}
	}),
```

I think the new `leaflet-draw-toolbar` [similarly extends](https://github.com/justinmanley/leaflet-draw-toolbar/blob/master/examples/popup.html#L37-L52) `LeafletToolbar.ToolbarAction` using `LeafletToolbar.ToolbarAction.extendOptions()` --

```js
				LeafletToolbar.ToolbarAction.extendOptions({
					toolbarIcon: { 
						className: 'leaflet-color-picker', 
						html: '<span class="fa fa-eyedropper"></span>' 
					},
					subToolbar: new LeafletToolbar({ actions: [
						L.ColorPicker.extendOptions({ color: '#db1d0f' }),
						L.ColorPicker.extendOptions({ color: '#025100' }),
						L.ColorPicker.extendOptions({ color: '#ffff00' }),
						L.ColorPicker.extendOptions({ color: '#0000ff' })
					]})
				})
```

This'll likely take some tweaking to get it working but there is documentation here: https://github.com/leaflet/Leaflet.Toolbar/wiki/Building-custom-toolbars

And I believe all the changes would be to this file: https://github.com/publiclab/Leaflet.DistortableImage/blob/master/src/edit/DistortableImage.EditToolbar.js

In addition to repointing all references to `leaflet-toolbar` to the new [leaflet-draw-toolbar](https://www.npmjs.com/package/leaflet-draw-toolbar), which is in NPM: https://github.com/publiclab/Leaflet.DistortableImage/search?utf8=%E2%9C%93&q=leaflet-toolbar&type=

**Comment by jywarren** [comment]
Actually apparently all we need is to update Leaflet.Toolbar to v0.3.x... trying in https://github.com/publiclab/Leaflet.DistortableImage/pull/79

**Comment by jywarren** [comment]
OK - got some ways in refactoring, as some was required for `v0.3.0` of Leaflet.Toolbar. The latest version does not produce errors. But it also does not produce a popup. 

This apparently needs further debugging! Any help appreciated.

**Comment by jywarren** [comment]
Making some good progress! Toolbar is created, but not displayed, in #79. May be that the actions are not getting added properly. Click events work, though.

**Comment by jywarren** [comment]
Fixed in #79
